src/core/ddsc/src/shm_monitor.c

```diff
@@ -80,7 +80,7 @@ dds_return_t shm_monitor_wake_and_enable(shm_monitor_t* monitor)
 dds_return_t shm_monitor_attach_reader(shm_monitor_t* monitor, struct dds_reader* reader) 
 {
 
-    if(iox_listener_attach_subscriber_event(monitor->m_listener, reader->m_iox_sub, SubscriberEvent_HAS_DATA, shm_subscriber_callback) != ListenerResult_SUCCESS) {
+    if(iox_listener_attach_subscriber_event(monitor->m_listener, reader->m_iox_sub, SubscriberState_HAS_DATA, shm_subscriber_callback) != ListenerResult_SUCCESS) {
```

Comment:
This looks weird to me: `iox_listener_attach_subscriber_event` and `SubscriberState_HAS_DATA`. Looking at the code and the enums, I would have expected `SubscriberEvent_DATA_RECEIVED` instead.

Reply:
Right, corrected via changing last commit and force-push.